lib/Sema/DerivedConformanceDistributedActor.cpp

```diff
@@ -582,11 +739,43 @@ deriveDistributedActorType_SerializationRequirement(
 
 ValueDecl *DerivedConformance::deriveDistributedActor(ValueDecl *requirement) {
   if (auto var = dyn_cast<VarDecl>(requirement)) {
-    if (var->getName() == Context.Id_id)
-      return deriveDistributedActor_id(*this);
+    ValueDecl *derivedValue = nullptr;
+    if (var->getName() == Context.Id_id) {
+      derivedValue = deriveDistributedActor_id(*this);
+    } else if (var->getName() == Context.Id_actorSystem) {
+      derivedValue = deriveDistributedActor_actorSystem(*this);
+    } else if (var->getName() == Context.Id_unownedExecutor) {
+      derivedValue = deriveDistributedActor_unownedExecutor(*this);
+    }
 
-    if (var->getName() == Context.Id_actorSystem)
-      return deriveDistributedActor_actorSystem(*this);
+    if (derivedValue) {
+      if (auto id = Nominal->getDistributedActorIDProperty()) {
+        if (auto system = Nominal->getDistributedActorSystemProperty()) {
+          if (auto classDecl = dyn_cast<ClassDecl>(Nominal)) {
+            if (auto unownedExecutor = classDecl->getUnownedExecutorProperty()) {
+              int idIdx, actorSystemIdx, unownedExecutorIdx = 0;
+              int idx = 0;
+              for (auto member : Nominal->getAllMembers()) {
```

Comment:
Seems like you could use `NominalTypeDecl::lookupDirect` instead of this loop.

Reply:
The purpose of this piece of code is to guarantee the order of fields is: `id, actorSystem` (which is ultra important for a distributed actor, it must be those two fields first otherwise IRGen and our allocations eventually will do very bad things) and then the executor.

A direct lookup would not give all the fields such that I can compare their order which is the purpose of this loop / assertion (I need to guard this with debug mode once I fix the bug on optimized builds here).

Reply:
Moved this under an `#ifndef NDEBUG` and documented it more